ddl/ddl_api.go

```diff
@@ -1029,6 +1025,27 @@ func (d *ddl) CreateTableWithLike(ctx sessionctx.Context, ident, referIdent ast.
 	return errors.Trace(err)
 }
 
+func buildTableInfoWithLike(d *ddl, ident ast.Ident, referTblInfo *model.TableInfo) (model.TableInfo, error) {
+	var err error
+	tblInfo := *referTblInfo
+	// No public column must in the last offset.
+	if tblInfo.Columns[len(tblInfo.Columns)-1].State != model.StatePublic {
```

Comment:
Please don't make assumptions and use an iteration to prune the non-public columns.

Reply:
done.